core/recipe/runner.go

```diff
@@ -30,192 +30,150 @@ func NewRunner(
 		monitor = new(defaultMonitor)
 	}
 	return &Runner{
-		extractorFactory:        extractorFactory,
+		extractorFactory: extractorFactory,
 		processorFactory: processorFactory,
 		sinkFactory:      sinkFactory,
 		monitor:          monitor,
 	}
 }
 
-func (r *Runner) Run(recipe Recipe) (err error) {
-	r.wg = new(sync.WaitGroup)
-	r.errChan = make(chan error)
-	getDuration := r.startDuration()
-	success := true
+func (r *Runner) Run(recipe Recipe) (run Run) {
+	var wg sync.WaitGroup
+	var (
+		getDuration = r.startDuration()
+		channel     = make(chan interface{})
+	)
+	run.Recipe = recipe
 
 	ctx, cancel := context.WithCancel(context.Background())
 	defer cancel()
+	// run extractors
+	{
```

Comment:
Any reason for this scope?

Reply:
no big reason, purely for readability. If it does not help I can remove it.

Reply:
Yes, I think it just makes the reader think and look for reasons and creates confusion.